File: AD-HOC/1329/main.dart
import 'dart:io';

void main() {
  int n;
  while ((n = int.parse(stdin.readLineSync()!)) != 0) {
    List<int> results = stdin.readLineSync()!.split(' ').map(int.parse).toList();
    int john = results.reduce((a, b) => a + b);
    int mary = n - john;
    print('Mary won $mary times and John won $john times');
  }
}

File: AD-HOC/1329/main.c
#include <stdio.h>

int main() {
    int n, r;
    scanf("%d", &n);
    while (n != 0) {
        int john = 0;
        for (int i = 0; i < n; i++) {
            scanf("%d", &r);
            john += r;
        }
        int mary = n - john;
        printf("Mary won %d times and John won %d times\n", mary, john);
        scanf("%d", &n);
    }
    return 0;
}
